plasmapy/particles/particle_class.py

```diff
@@ -421,164 +421,182 @@ class Particle(AbstractPhysicalParticle):
-                raise InvalidParticleError(errmsg) from exc
+        attributes["atomic number"] = Element["atomic number"]
+        attributes["element name"] = Element["element name"]
 
-            for key in nomenclature.keys():
-                attributes[key] = nomenclature[key]
+        # Set the lepton number to zero for elements, isotopes, and
+        # ions.  The lepton number will probably come up primarily
+        # during nuclear reactions.
 
-            element = attributes["element"]
-            isotope = attributes["isotope"]
-            ion = attributes["ion"]
+        attributes["lepton number"] = 0
 
-            if element:
-                categories.add("element")
-            if isotope:
-                categories.add("isotope")
-            if self.element and self._attributes["charge number"]:
-                categories.add("ion")
+        if isotope:
 
-            # Element properties
+            Isotope = _isotopes[isotope]
 
-            Element = _elements[element]
+            attributes["baryon number"] = Isotope["mass number"]
+            attributes["isotope mass"] = Isotope.get("mass", None)
+            attributes["isotopic abundance"] = Isotope.get("abundance", 0.0)
 
-            attributes["atomic number"] = Element["atomic number"]
-            attributes["element name"] = Element["element name"]
+            if Isotope["stable"]:
+                attributes["half-life"] = np.inf * u.s
+            else:
+                attributes["half-life"] = Isotope.get("half-life", None)
 
-            # Set the lepton number to zero for elements, isotopes, and
-            # ions.  The lepton number will probably come up primarily
-            # during nuclear reactions.
+        if element and not isotope:
+            attributes["standard atomic weight"] = Element.get("atomic mass", None)
 
-            attributes["lepton number"] = 0
+        if ion in _special_ion_masses.keys():
+            attributes["mass"] = _special_ion_masses[ion]
 
-            if isotope:
+        attributes["periodic table"] = _PeriodicTable(
+            group=Element["group"],
+            period=Element["period"],
+            block=Element["block"],
+            category=Element["category"],
+        )
 
-                Isotope = _isotopes[isotope]
+        categories.add(Element["category"])
 
-                attributes["baryon number"] = Isotope["mass number"]
-                attributes["isotope mass"] = Isotope.get("mass", None)
-                attributes["isotopic abundance"] = Isotope.get("abundance", 0.0)
+    def _add_charge_information(self):
+        """Assign attributes and categories related to charge information."""
+        if self._attributes["charge number"] == 1:
+            self._attributes["charge"] = const.e.si
+        elif self._attributes["charge number"] is not None:
+            self._attributes["charge"] = self._attributes["charge number"] * const.e.si
```

Comment:
```suggestion
            self._attributes["charge"] = self._attributes["charge number"] * const.e.si
            
```
whitespace for visibility